### DISCUSSION TITLE
Do pipeline components affect training accuracy?

### DISCUSSION THREAD

**Discussion by nick-antisoc** [explosion/spaCy#7663] [Help: Other Questions] [answered]
I would like some info on how different components depend on each other during training.

Let's say for example I want to train a text categorizer `textcat`. 
If I add to the pipeline a `ner` component, will this addition improve the final accuracy? 

Does the trainable component use the ones before it during training in a way that would result to better accuracy?
Do only vectors/pretraining affect accuracy or the actual pipeline plays also an important role here?

Thanks,
Nick

**Comment by polm** [comment, marked as answer]
This is actually something we're working on making clearer at the moment, but in general pipeline components don't affect each other's accuracy. 

1. Components generally listen to a tok2vec or transformer layer further up the pipeline, which is the main determiner of accuracy besides the model itself.
2. Components *are* potentially influenced by changes to the doc object if they modify the same fields; this comes up in NER, where it's common to have a rule-based system and a statistical one and their order matters. 
3. Components can have internal models that aren't in the pipeline general. In the pretrained pipelines NER uses a separate tok2vec from the pipeline one. 

In your case of textcat and NER: no, the NER annotations are not going to affect textcat. 

> Do only vectors/pretraining affect accuracy or the actual pipeline plays also an important role here?

For textcat, yes, basically only the input representations would make a difference.
